src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+        }
+ 
+        // verify that no new address was created
+        listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
+        system.assertEquals(2, listAddr.size());
+        for (Address__c addr : listAddr) {
+            system.assert(addr.MailingStreet__c.equals('direct street  edit'));
+        }                   
+    }
+
+    /*********************************************************************************************************
+    @description Update accounts' billing address to cause an update of the default Address object.  
+    verify:
+        existing Default Address updated
+        Account get updated Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrExisting() {
+        configSettings();
+        
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // modify some of the account addresses directly
+        // NOTE: we only modify 1 field, so it will be treated as an update to an existing address!
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = 'Direct Street Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;
+
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;
+
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = ' direct street  edit ';    // whitespace and casing should not count as real edit.
+            accsAddrs.accs[i].BillingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('direct street  edit'));
+            system.assert(acc.BillingState.contains('Oregon')); 
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.